### PR TITLE
publish button greyed out unless add reading form complete

### PR DESCRIPTION
**Pull request by darrenmason**
tested with @state.id.  datetimepicker can be emptied after population but retains last value.

### CODE REVIEW

**Comment by darrenmason** [comment]
image

File: src/components/task-plan/reading.cjsx [hunk 1]
```diff
@@ -116,10 +116,20 @@ ReadingPlan = React.createClass
 
   render: ->
     id = @getParams().id
-    footer = <BS.Button bsStyle="primary" onClick={@publish}>Publish</BS.Button>
+
+    if id
+      plan = TaskPlanStore.get(id)
```
Comment (philschatz) [inline src/components/task-plan/reading.cjsx]: is `@state.id` ever not the same as `id`?

File: src/components/task-plan/reading.cjsx [hunk 2]
```diff
@@ -116,10 +116,20 @@ ReadingPlan = React.createClass
...
     id = @getParams().id
-    footer = <BS.Button bsStyle="primary" onClick={@publish}>Publish</BS.Button>
+
+    if id
+      plan = TaskPlanStore.get(id)    
+    else
+      plan = TaskPlanStore.get(@state.id)
+     
+    if plan.title and plan.due_at and plan.settings.page_ids.length > 0
+    then btnDisable = '' else btnDisable = 'disabled'
```
Comment (philschatz) [inline src/components/task-plan/reading.cjsx]: FYI: There is already an existing syntax in tutor-js for optionally adding classes: https://github.com/openstax/tutor-js/blob/master/src/components/question.cjsx#L36

Comment (darrenmason) [inline src/components/task-plan/reading.cjsx]: ok I will update syntax.  Also I noticed this module may be helpful : https://github.com/JedWatson/classnames

It was originally here : http://facebook.github.io/react/docs/class-name-manipulation.html

Comment (philschatz) [inline src/components/task-plan/reading.cjsx]: Yep, want to make a separate Pull request that changes all `className={classes.join(' ')}` to use `classes`?

File: src/components/task-plan/reading.cjsx (hunk 1 above)
Comment (darrenmason) [inline src/components/task-plan/reading.cjsx]: console.log on initial load in render:

getParams().id: undefined
@state.id: _CREATING_0
getParams().id: undefined
@state.id: _CREATING_0

it seems getParams() checks only the location.href?

Comment (philschatz) [inline src/components/task-plan/reading.cjsx]: Yep, that's correct. `@getParams()` only pulls from the `window.location`.
